rest/data/aliases.go

```diff
@@ -107,6 +108,49 @@ func (d *DBAliasConnector) UpdateProjectAliases(projectId string, aliases []rest
 	return catcher.Resolve()
 }
 
+// should be called in the resolver
+func (pc *DBAliasConnector) UpdateAliasesForSection(projectId string, updatedAliases []restModel.APIProjectAlias,
+	originalAliases []model.ProjectAlias, section model.ProjectRefSection) (bool, error) {
+	aliasesIdMap := map[string]bool{}
+	aliasesToUpdate := []restModel.APIProjectAlias{}
+
+	for _, a := range updatedAliases {
+		if shouldSkipAliasForSection(section, utility.FromStringPtr(a.Alias)) {
+			continue
+		}
+		aliasesToUpdate = append(aliasesToUpdate, a)
+		aliasesIdMap[utility.FromStringPtr(a.ID)] = true
+	}
+	if err := pc.UpdateProjectAliases(projectId, aliasesToUpdate); err != nil {
+		return false, errors.Wrap(err, "error updating project aliasesToUpdate")
+	}
+	catcher := grip.NewBasicCatcher()
+	// delete any aliasesToUpdate that were in the list before but are not now
+	for _, originalAlias := range originalAliases {
```

Comment:
what does this loop do? where is it removing it from and why is it necessary after UpdateProjectAliases()?

Reply:
This deletes any aliases that have been deleted by the user, i.e. they existed originally (in the before ProjectSettings struct) but don't exist anymore. It's removing those aliases from the alias collection.

Reply:
ah